src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1647,6 +1647,83 @@ class DropMaterializedViewStatementSegment(BaseSegment):
     )
 
 
+@postgres_dialect.segment()
+class AlterViewStatementSegment(BaseSegment):
+    """A `ALTER VIEW` statement.
+
+    As specified in https://www.postgresql.org/docs/14/sql-alterview.html
+    """
+
+    type = "alter_view_statement"
+
+    match_grammar = StartsWith(Sequence("ALTER", "VIEW"))
+
+    parse_grammar = Sequence(
+        "ALTER",
+        "VIEW",
+        Ref("IfExistsGrammar", optional=True),
+        Ref("TableReferenceSegment"),
+        OneOf(
+            Sequence(
+                "ALTER",
+                Ref.keyword("COLUMN", optional=True),
+                Ref("ColumnReferenceSegment"),
+                OneOf(
+                    Sequence(
+                        "SET",
+                        "DEFAULT",
+                        OneOf(
+                            Ref("LiteralGrammar"),
+                            Ref("FunctionSegment"),
+                            Ref("BareFunctionSegment"),
+                            Ref("ExpressionSegment"),
+                        ),
+                    ),
```

Comment:
You're correct here. I noticed that the existing [AlterTableActionSegment](https://github.com/sqlfluff/sqlfluff/blob/7aecf672872d05b608d614dc1f2b040e17a4f902/src/sqlfluff/dialects/dialect_postgres.py#L1275) only allows expression segments when it should also account for the segments you have here. If you're interested would you be able to create a separate PR to match that line to the SET DEFAULT grammar you have here and add the corresponding unit tests? 😄

Reply:
Raised here: https://github.com/sqlfluff/sqlfluff/pull/2101